packages/backend/src/startup.rs

```
@@ -0,0 +1,98 @@
+    net::{IpAddr, TcpListener},
+    sync::Arc,
+};
+
+use actix_cors::Cors;
+use actix_web::{
+    dev::Server,
+    web::{self, ServiceConfig},
+    App, HttpResponse, HttpServer, Responder,
+};
+use lost_pixel::{client::LostPixelClient, docker::DockerClient};
+
+use crate::{
+    routes::{generate, get_story, list_stories, promote, spa},
+    settings::Settings,
+};
+
+pub struct Application {
+    ip: IpAddr,
+    port: u16,
+    server: Server,
+}
+
+impl Application {
+    pub fn build(settings: Settings) -> Result<Self, std::io::Error> {
+        let address = format!("{}:{}", settings.host, settings.port);
+        let listener = TcpListener::bind(address)?;
+        let socket_addr = listener.local_addr()?;
+        let ip = socket_addr.ip();
+        let port = socket_addr.port();
+
+        let configurator = configure_app(&settings);
+
+        let server = HttpServer::new(move || {
+            App::new()
+                .wrap(
+                    Cors::default()
+                        .allow_any_header()
+                        .allow_any_method()
+                        .allow_any_origin(),
+                )
+                .configure(configurator.clone())
+        })
+        .listen(listener)?
+        .run();
+
+        Ok(Self { ip, port, server })
+    }
+
+    pub fn ip(&self) -> IpAddr {
+        self.ip
+    }
+
+    pub fn port(&self) -> u16 {
+        self.port
+    }
+
+    pub async fn run(self) -> Result<(), std::io::Error> {
+        self.server.await
+    }
+}
+
+async fn pong() -> impl Responder {
+    HttpResponse::Ok().body("pong")
+}
+
+/// # Panics
+pub fn get_lost_pixel_client(settings: &Settings) -> Arc<LostPixelClient> {
+    let docker_client = DockerClient {};
+    let docker_client = Box::new(docker_client);
+    let client = LostPixelClient::init(
+        settings.workspace.clone(),
+        &settings.lost_pixel_image_tag,
+        docker_client,
+    )
+    .expect("Failed to initialize LostPixelClient");
```

Why you panic instead of returning Option?

---

I was motivated by the logic that this is an unrecoverable error. The backend cannot work without LostPixelClient.

---

Ah, I only now noticed that this is purely web app code, you may ignore my comment below.